src/main/java/com/github/simonpercic/responseecho/MainController.java

```diff
@@ -64,27 +70,53 @@ ModelAndView responseInfo(
             @PathVariable("response") String responseBodyString,
             @RequestParam(value = SharedConstants.QUERY_PARAM_REQUEST_BODY, required = false) String requestBodyString,
             @RequestParam(value = SharedConstants.QUERY_PARAM_DATA, required = false) String logDataString,
-            @RequestParam(value = SharedConstants.QUERY_SHORTEN_URL, required = false) boolean shortenUrl)
+            @RequestParam(value = SharedConstants.QUERY_SHORTEN_URL, required = false) boolean shortenUrl,
+            @RequestParam(value = SharedConstants.QUERY_FIREBASE_URL, required = false) boolean firebaseUrl)
             throws IOException {
+        HttpUrl infoUrl = null;
+        HttpUrl.Builder infoUrlBuilder;
 
-        HttpUrl.Builder infoUrlBuilder = requestHttpUrl(request)
-                .addPathSegment(Constants.V1)
-                .addPathSegment(RESPONSE_INFO)
-                .addEncodedPathSegment(responseBodyString);
+        if (firebaseUrl) {
+            String data = "";
+            data = FirebaseUtils.INSTANCE.getData(responseBodyString);
+            String url = requestHttpUrl(request)
```

Comment:
It seems like paths of firebaseUrl and not-firebaseUrl can be reused. They seem to be somewhat duplicated now.